openmc/material.py

```diff
@@ -282,15 +275,49 @@ def fissionable_mass(self) -> float:
 
     @property
     def decay_photon_energy(self) -> Optional[Univariate]:
-        atoms = self.get_nuclide_atoms()
+        warnings.warn(
+            "The 'decay_photon_energy' property has been replaced by the "
+            "get_decay_photon_energy() method and will be removed in a future "
+            "version.", FutureWarning)
+        return self.get_decay_photon_energy(0.0)
+
+    def get_decay_photon_energy(self, clip_tolerance: float = 1e-6) -> Optional[Univariate]:
+        """Return energy distribution of decay photons from unstable nuclides.
+
+        Parameters
+        ----------
+        clip_tolerance : float
+            Maximum fraction of integral of the product of `x` and `p` for
+            discrete distributions that will be discarded.
+
+        Returns
+        -------
+        Decay photon energy distribution. The integral of this distribution is
+        the total intensity of the photon source in [Bq].
+
+        """
         dists = []
         probs = []
-        for nuc, num_atoms in atoms.items():
+        for nuc, num_atoms in self.get_nuclide_atoms().items():
```

Comment:
What were your thoughts on changing this to `get_nuclide_atom_densities()` and using a volume multiplier (either 1 for not set or `self.volume`)?
It would be useful to compare spectra from compositions on a per cc basis.

Reply:
Just to make sure I'm understanding, are you suggesting that we add an argument that allows the user to ask for different units, something like `mat.get_decay_photon_energy(units='Bg/cc')`?

Reply:
I don't typically like those types of arguments, but in this particular case that might be the best way. Although an alternative is for a volume kwarg like `volume=None` that overrides `self.volume`. That way you can pass in `volume=1` to get `Bq/cc` implicitly rather than asking to futz around with unit conversions. I think I actually like this approach more. We can still error out if `volume is None and self.volume is None` rather than silently changing units on the user. You always get units of `Bq`, but you can also get `Bq/cc` without modifying the original material object _and_ without having to specify a volume for your material in the first place. What do you think?

Reply:
For other methods (`get_activity`, `get_decay_heat`), we have both `units` and `volume` arguments so my preference would be to replicate those. OK with you?

Reply:
yeah that works